src/mlpack/methods/ann/layer/linear.hpp

```diff
@@ -124,8 +124,15 @@ class Linear
   //! Modify the gradient.
   OutputDataType& Gradient() { return gradient; }
 
+  //! Get the weight of the layer.
+  InputDataType const& Weight() const { return weight; }
+  //! Modify the weight of the layer.
+  InputDataType& Weight() { return weight; }
+
+  //! Get the bias of the layer.
+  InputDataType const& Bias() const { return bias; }
   //! Modify the bias weights of the layer.
-  arma::mat& Bias() { return bias; }
+  InputDataType& Bias() { return bias; }
```

Comment:
Good point, there is no reason, no.

Reply:
Actually, it should have been `OutputDataType` since `weight` is of `OutputDataType`, sorry for that, I will correct it in next commit. In fact templating is a concern in all-around `ann` codebase. It will be a lot of work so I thought of handling it in a separate PR.
Is that okay for now?

Reply:
As long as you do it, it's fine by me. Doesn't matter if it is in this PR or a separate one. 
Maybe you would like to create an issue for now, listing this templating problem and then later we can resolve it with a separate PR?
Let's also wait for zoq's opinion on this.